sdk/nodejs/cmd/run/run.ts

```diff
@@ -59,6 +61,28 @@ function packageObjectFromProjectRoot(projectRoot: string): Record<string, any>
     }
 }
 
+// Reads and parses the contents of .npmrc file if it exists under the project root
+// This assumes that .npmrc is a sibling to package.json
+function npmRcFromProjectRoot(projectRoot: string): Record<string, any>  {
+    const emptyConfig = {};
+    try {
+        const npmRcPath = path.join(projectRoot, ".npmrc");
+        if (!fs.existsSync(npmRcPath)) {
+            return emptyConfig;
+        }
+        // file .npmrc exists, read its contents
+        const npmRc = fs.readFileSync(npmRcPath, "utf-8");
+        // Use ini to parse the contents of the .npmrc file
+        // This is what node does as described in the npm docs
+        // https://docs.npmjs.com/cli/v8/configuring-npm/npmrc#comments
+        return ini.parse(npmRc);
+    } catch {
```

Comment:
Wonder if we should just allow this error to bubble up. It's probably something like bad permissions and having that explict error might be better than silently ignoring if engine-strict is set?

Reply:
I followed the same approach as with `packageObjectFromProjectRoot` where we just return an empty object if we are unable to read or parse it. Do you think it is likely that an error would occur when we are only _reading_ the file?

Reply:
Only errors I can think likely are permission issues. It's probably rare enough to not really worry either way.

Reply:
Would we have a way of raising a warning here instead - to help the user if it is some kind of permissions issue?